src/dc_mimefactory.c

```diff
@@ -526,6 +527,13 @@ int dc_mimefactory_render(dc_mimefactory_t* factory)
 			}
 		}
 
+		// beside key- and member-changes, force re-gossip every 48 hours
+		#define AUTO_REGOSSIP (2*24*60*60)
+		if (chat->gossiped_timestamp==0
+		 || chat->gossiped_timestamp+AUTO_REGOSSIP < time(NULL) ) {
```

Comment:
maybe ```(time(NULL) - chat->gossiped_timestamp) < AUTO_REGOSSIP``` would be a more understandable way to describe the logic ;)

Reply:
do you mean `(time(NULL) - chat->gossiped_timestamp) > AUTO_REGOSSIP` ?

these things are hard ... ;)